### PR TITLE
Fix bool parameter marshalling in Microsoft.DiaSymReader interfaces

### PR DESCRIPTION
**Pull request by tmat**

### CODE REVIEW

**Comment by tmat** [comment]
@gregg-miskelly @pharring

**Comment by pharring** [comment]
This looks OK to me, assuming the parameters are actually specified as a Windows BOOL in the IDL (as opposed to a VARIANT_BOOL).

File: src/Dependencies/Microsoft.DiaSymReader/ISymUnmanagedAsyncMethod.cs [hunk 1]
```diff
@@ -12,13 +12,13 @@ namespace Microsoft.DiaSymReader
     public interface ISymUnmanagedAsyncMethod
     {
         [PreserveSig]
-        int IsAsyncMethod(out bool value);
+        int IsAsyncMethod([MarshalAs(UnmanagedType.Bool)]out bool value);
```
Comment (pharring) [inline src/Dependencies/Microsoft.DiaSymReader/ISymUnmanagedAsyncMethod.cs]: Is it possible to remove ```[PreserveSig]``` on these methods? It's certainly desirable to make it easier for managed callers. There'd be two reasons not to do that:
1. We are trying to match another interop definition precisely (or this _is_ the interop assembly for others to consume)
2. The HRESULTs returned are non-standard (i.e. anything other than the usual S_OK, E_INVALIDARG, E_OUTOFMEMORY, E_POINTER, etc.)

Comment (pharring) [inline src/Dependencies/Microsoft.DiaSymReader/ISymUnmanagedAsyncMethod.cs]: Note that, if you do remove the [PreserveSig], for this case you would have to annotate the return type with ```[return: MarshalAs(UnmanagedType.Bool)]```

**Comment by gregg-miskelly** [comment]
:+1:

File: src/Dependencies/Microsoft.DiaSymReader/ISymUnmanagedAsyncMethod.cs (hunk 1 above)
Comment (gregg-miskelly) [inline src/Dependencies/Microsoft.DiaSymReader/ISymUnmanagedAsyncMethod.cs]: This is effectively the PIA definition for these interfaces -- this is the definition that one must use if consuming or implementing these interfaces from managed code in Visual Studio.

**Comment by tmat** [comment]
test win please

File: src/Dependencies/Microsoft.DiaSymReader/ISymUnmanagedAsyncMethod.cs (hunk 1 above)
Comment (tmat) [inline src/Dependencies/Microsoft.DiaSymReader/ISymUnmanagedAsyncMethod.cs]: @pharring Yes, all are declared as BOOL in IDL.

**Pull request merged**